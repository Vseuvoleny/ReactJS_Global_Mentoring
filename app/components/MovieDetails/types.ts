import { IFilm } from "../../types/types";

export interface IMovieDetails {
    film: IFilm;
}