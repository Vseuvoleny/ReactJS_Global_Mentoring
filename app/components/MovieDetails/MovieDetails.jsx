import React, { memo } from "react";
import PropTypes from "prop-types";
import "./MovieDetails.module.scss";

const MovieDetails = memo(({ img, title, rating, year, runtime, overview }) => {
  return (
    <section className="movie-details">
      <picture className="movie-details__image">
        <img src={img} alt="" width={323} height={486} />
      </picture>
      <section className="movie-details__description">
        <div className="movie-details__title">
          <span className="name">{title}</span>
          <span className="rate">
            <span>{rating}</span>
          </span>
        </div>
        <div className="movie-details__genre">{"Action & Adventure"}</div>
        <div className="movie-details__year">
          <span className="year">{year}</span>
          <span className="timeing">{runtime}</span>
        </div>
        <div className="movie-details__plot">{overview}</div>
      </section>
    </section>
  );
});

MovieDetails.propTypes = {
  title: PropTypes.string,
  img: PropTypes.string,
};

export default MovieDetails;
