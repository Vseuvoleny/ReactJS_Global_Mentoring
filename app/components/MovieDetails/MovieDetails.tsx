import React from "react";
import classes from "./MovieDetails.module.scss";

const MovieDetails = ({ film }) => {
  const { poster_path, title, vote_average, year, runtime, overview, genres } =
    film;

  return (
    <section className={classes.movie_details}>
      <picture className={classes.movie_details__image}>
        <img src={poster_path} alt={title} width={323} height={486} />
      </picture>
      <section className={classes.movie_details__description}>
        <div className={classes.movie_details__title}>
          <span className={classes.name}>{title}</span>
          <span className={classes.rate}>
            <span>{vote_average}</span>
          </span>
        </div>
        <div className={classes.movie_details__genre}>{genres.join(", ")}</div>
        <div className={classes.movie_details__year}>
          <span className={classes.year}>{year}</span>
          <span className={classes.timeing}>{runtime}</span>
        </div>
        <div className={classes.movie_details__plot}>{overview}</div>
      </section>
    </section>
  );
};

export default MovieDetails;
