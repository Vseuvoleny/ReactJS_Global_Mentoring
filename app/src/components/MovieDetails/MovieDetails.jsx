import React from "react";
import PropTypes from "prop-types";

const MovieDetails = ({ title = "Pulp Fiction" }) => {
  return (
    <section className="movie-details">
      <picture className="movie-details__image">
        <img src="" alt="" />
      </picture>
      <section className="movie-details__description">
        <div>
          <span>{title}</span>
          <span> 8.9</span>
        </div>
        <div className="movie-genre">{"Action & Adventure"}</div>
        <div>
          <span>1994</span>
          <span>2h 34min</span>
        </div>
        <div>
          Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta)
          are two hit men who are out to retrieve a suitcase stolen from their
          employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also
          asked Vincent to take his wife Mia (Uma Thurman) out a few days later
          when Wallace himself will be out of town. Butch Coolidge (Bruce
          Willis) is an aging boxer who is paid by Wallace to lose his fight.
          The lives of these seemingly unrelated people are woven together
          comprising of a series of funny, bizarre and uncalled-for
          incidents.â€”Soumitra
        </div>
      </section>
    </section>
  );
};

MovieDetails.propTypes = {
  title: PropTypes.string,
};

export default MovieDetails;
