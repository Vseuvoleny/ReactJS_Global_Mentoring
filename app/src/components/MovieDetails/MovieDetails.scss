.movie-details {
  display: flex;
  margin-bottom: 30px;
  &__image {
    margin-right: 60px;
  }
  &__description {
    max-width: 698px;
    width: 100%;
  }
  &__title {
    display: flex;
    align-items: center;
    column-gap: 25px;
    & > * {
      display: inline-flex;
    }
    .name {
      font-weight: 300;
      font-size: 40px;
      line-height: 49px;
      letter-spacing: 1px;
      text-transform: uppercase;
    }
    .rate {
      border: 1px solid #ffffff;
      width: 60px;
      height: 60px;
      border-radius: 50%;
      position: relative;
      span {
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
      }
    }
  }
  &__genre {
    font-weight: 500;
    font-size: 14px;
    line-height: 17px;
    opacity: 0.5;
  }
  &__year {
    margin: 30px 0;
    & > * {
      font-weight: 300;
      font-size: 24px;
      line-height: 29px;
      color: #f65261;
      display: inline-block;
    }
    .year {
      margin-right: 51px;
    }
  }
  &__plot {
    opacity: 0.5;
    font-weight: 300;
    font-size: 20px;
    line-height: 24px;
  }
}
