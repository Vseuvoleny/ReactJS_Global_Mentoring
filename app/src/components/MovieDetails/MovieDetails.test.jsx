import { render } from "@testing-library/react";
import React from "react";
import MovieDetails from "./MovieDetails";

describe("MovieDetails component", () => {
  test("should render component", () => {
    const movie = {
      img: "./movies.jpeg",
      title: "Movie title",
      rating: "5",
      year: "2011",
      runtime: "120",
      overview: "Movie overwiev",
    };

    const { container, getByText, getByAltText } = render(
      <MovieDetails {...movie} />
    );
    expect(container).toMatchSnapshot();
    expect(getByText("Movie title")).toBeInTheDocument();
    expect(getByText("5")).toBeInTheDocument();
    expect(getByText("120")).toBeInTheDocument();
    expect(getByText("Movie overwiev")).toBeInTheDocument();
    expect(getByAltText("Movie title")).toBeInTheDocument();
  });
});
