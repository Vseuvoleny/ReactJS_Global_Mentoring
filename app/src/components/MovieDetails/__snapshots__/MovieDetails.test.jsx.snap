// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component should render component 1`] = `
<div>
  <section
    class="movie-details"
  >
    <picture
      class="movie-details__image"
    >
      <img
        alt="Movie title"
        height="486"
        src="./movies.jpeg"
        width="323"
      />
    </picture>
    <section
      class="movie-details__description"
    >
      <div
        class="movie-details__title"
      >
        <span
          class="name"
        >
          Movie title
        </span>
        <span
          class="rate"
        >
          <span>
            5
          </span>
        </span>
      </div>
      <div
        class="movie-details__genre"
      >
        Action & Adventure
      </div>
      <div
        class="movie-details__year"
      >
        <span
          class="year"
        >
          2011
        </span>
        <span
          class="timeing"
        >
          120
        </span>
      </div>
      <div
        class="movie-details__plot"
      >
        Movie overwiev
      </div>
    </section>
  </section>
</div>
`;
